server/chat-plugins/the-studio.ts

```diff
@@ -0,0 +1,613 @@
+		}
+		if (!videoInfo && !suggested) {
+			buf += `<br /><strong>Score:</strong> ${Chat.count(rec.likes, "points")}`;
+		}
+		if (!rec.userData.avatar) {
+			buf += `<br /><strong>Recommended by:</strong> ${rec.userData.name}`;
+		}
+		buf += `<hr />`;
+		if (suggested) {
+			buf += Utils.html`<button class="button" name="send" value="/approvesuggestion ${rec.userData.name}|${rec.artist}|${rec.title}">Approve</button> | `;
+			buf += Utils.html`<button class="button" name="send" value="/denysuggestion ${rec.userData.name}|${rec.artist}|${rec.title}">Deny</button>`;
+		} else {
+			buf += Utils.html`<button class="button" name="send" value="/likerec ${rec.artist}|${rec.title}" style="float:right;display:inline;padding:3px 5px;font-size:8pt;">`;
+			buf += `<img src="https://${Config.routes.client}/sprites/bwicons/441.png" style="margin:-9px 0 -6px -7px;" width="32" height="32" />`;
+			buf += `<span style="position:relative;bottom:2.6px;">Upvote</span></button>`;
+		}
+		buf += `</td>`;
+		if (!suggested && rec.userData.avatar) {
+			buf += `<td style="text-align:center;width:110px;background:rgba(255,255,255,0.4);border-radius:15px;">`;
+			const isCustom = rec.userData.avatar.startsWith('#');
+			buf += `<img style="margin-bottom:-38px;" src="https://${Config.routes.client}/sprites/trainers${isCustom ? '-custom' : ''}/${isCustom ? rec.userData.avatar.slice(1) : rec.userData.avatar}.png" width="80" height="80" />`;
+			buf += `<br /><span style="background:rgba(0,0,0,0.5);padding:1.5px 4px;color:white;font-size:7pt;">Recommended by:`;
+			buf += `<br /><strong>${rec.userData.name}</strong></span></td>`;
+		}
+		buf += `</tbody></table>`;
+		buf += `</div>`;
+		return buf;
+	}
+
+	likeRecommendation(artist: string, title: string, liker: User) {
+		const rec = this.get(artist, title);
+		if (!rec) {
+			throw new Chat.ErrorMessage(`The song titled '${title}' by ${artist} isn't recommended.`);
+		}
+		if (!rec.liked) {
+			rec.liked = {ips: [], userids: []};
+		}
+		if (rec.liked.ips.includes(liker.latestIp) || rec.liked.userids.includes(liker.id)) {
+			throw new Chat.ErrorMessage(`You've already liked this recommendation.`);
+		}
+		rec.likes++;
+		rec.liked.ips.push(liker.latestIp);
+		rec.liked.userids.push(liker.id);
+		saveRecommendations();
+	}
+
+	get(artist: string, title: string, submitter?: string, fromSuggestions = false) {
+		let recs = recommendations.saved;
+		if (fromSuggestions) recs = recommendations.suggested;
+		return recs.find(x => toID(x.artist) === toID(artist) &&
+			toID(x.title) === toID(title) &&
+			(submitter ? toID(x.userData.name) === toID(submitter) : true));
+	}
+
+	getIndex(artist: string, title: string, submitter?: string, fromSuggestions = false) {
+		let recs = recommendations.saved;
+		if (fromSuggestions) recs = recommendations.suggested;
+		const index = recs.findIndex(x => toID(x.artist) === toID(artist) &&
```

Comment:
This is not easy to read. I'd suggest refactoring it into a full arrow function (with curly braces and a `return`).

Reply:
the new eslint rules forbade that

Reply:
it was originally like that, there's even a commit by me to change it to this because of it

Reply:
https://github.com/smogon/pokemon-showdown/pull/7542/commits/fd8777b096a2790495b3d5249a176e9cd966398d

Reply:
Hm, okay. I thought it would be permissible if there were multiple return statements (e.g. `if (!submitter) return true;`).

Reply:
You can still make it more readable.